stl/inc/xutility

```diff
@@ -1924,24 +1924,28 @@ public:
         requires (!is_same_v<_Other, _BidIt>) && convertible_to<const _Other&, _BidIt>
             && assignable_from<_BidIt&, const _Other&>
 #endif // __cpp_lib_concepts
-    _CONSTEXPR17 reverse_iterator& operator=(const reverse_iterator<_Other>& _Right) {
+    _CONSTEXPR17 reverse_iterator& operator=(const reverse_iterator<_Other>& _Right) noexcept(
+        is_nothrow_constructible_v<_BidIt, const _Other&>) /* strengthened */ {
         current = _Right.current;
         return *this;
     }
     // clang-format on
 
-    _NODISCARD _CONSTEXPR17 _BidIt base() const {
+    _NODISCARD _CONSTEXPR17 _BidIt base() const noexcept(is_nothrow_copy_constructible_v<_BidIt>) /* strengthened */ {
         return current;
     }
 
-    _NODISCARD _CONSTEXPR17 reference operator*() const {
+    _NODISCARD _CONSTEXPR17 reference operator*() const noexcept(is_nothrow_copy_constructible_v<_BidIt>&& noexcept(
+        _Implicitly_convert_to<reference>(*--(_STD declval<_BidIt&>())))) /* strengthened */ {
         _BidIt _Tmp = current;
         return *--_Tmp;
     }
 
 #ifdef __cpp_lib_concepts
     // clang-format off
-    _NODISCARD constexpr pointer operator->() const
+    _NODISCARD constexpr pointer operator->() const noexcept(is_nothrow_copy_constructible_v<_BidIt>
+        && noexcept(--(_STD declval<_BidIt&>())) && noexcept(_Implicitly_convert_to<pointer>(
+            _Operator_arrow(_STD declval<_BidIt&>(), is_pointer<_BidIt>())))) /* strengthened */

```

Comment:
I'll leave this outdated comment open since it still sort-of applies, even if I don't think the suggestion is a clear win. Consider and resolve as you like.